### ISSUE TITLE
Test Vectors - Verifiable Credentials - JSON Schema

### ISSUE THREAD

**Issue by decentralgabe** [TBD54566975/web5-spec#170]
This issue outlines gaps in testing of JSON Schema independently along with with its usage in conjunction with the W3C VCDM v1.1. Tasks may have partial test coverage today (linked if true). If sufficient test coverage already exists, please close the task and link the relevant test.

Tasks may need discussion on the best way to approach them. The goal is to create a set of language agnostic test vectors that give us sufficient confidence in maintaining interoperability of features between implementations.

## JSON Schema

https://tbd54566975.github.io/web5-spec/#json-schema

- [ ] Create JSON Schemas against Draft 7, 2020-12
- [ ] Validate data against JSON Schemas Draft 7, 2020-12

_Note this is generic JSON Schema functionality_

## VC JSON Schema

https://tbd54566975.github.io/web5-spec/#credential-schema

- [ ] Create a VC JSON Schema of type `JsonSchema` in a VC
- [ ] Resolve a JSON Schema from a web resource given its id
- [ ] Validate a given VC against a VC JSON Schema of type `JsonSchema`

**Comment by nitro-neal** [comment]
Would we host some real life json schemas? We can just put some examples in the web5-spec github?

**Comment by decentralgabe** [comment]
yes, take a look at the test vectors here for a sense of what's covered -- https://github.com/w3c/vc-json-schema-test-suite/blob/main/tests/input/jsonschema/2020-12/1-schema.json

note, we do not need to do web-based resolution, this can be completely static. but we should have schemas that are fairly robust
